webui/settings.py

```diff
@@ -165,62 +165,61 @@
 
 class ConfigFile:
     __instance = None
-    __file_hash = None
-    filename = None
-    config = None
 
-    def __new__(cls, filename=None):
+    def __new__(cls, filename: str = None):
         if ConfigFile.__instance is None:
             ConfigFile.__instance = object.__new__(cls)
-        ConfigFile.__instance.filename = filename or CONFIG_FILE
+            ConfigFile.__instance.filename = filename or CONFIG_FILE
+            ConfigFile.__instance.config = configparser.ConfigParser()
+            ConfigFile.__instance.config.read(ConfigFile.__instance.filename)
         return ConfigFile.__instance
 
-    def get_hash(self):
-        with open(self.filename, 'r') as f:
-            h = hashlib.sha256()
-            h.update(f.read().encode('utf-8'))
-            return h.hexdigest()
-        return None
-
-    def check_file(self):
-        file_hash = self.get_hash()
-        if self.__file_hash is None or self.__file_hash != file_hash:
-            self.__file_hash = file_hash
-            self.config = configparser.ConfigParser()
-            self.config.read(self.filename)
-
-    def get(self, name, default=None):
-        self.check_file()
-        d = self.config
-        if not isinstance(name, list):
-            name = [name]
-        for i in name:
-            if i in d:
-                d = d[i]
+    def get(self, config_path: str, default: str = None) -> str:
+        config_pointer = self.config
+        config_array = config_path.split('/')
+        for i in config_array:
+            if i in config_pointer:
+                config_pointer = config_pointer[i]
             else:
                 if default is None:
-                    raise LookupError('Missing attribute {} in file {}'.format('.'.join(name), self.filename))
+                    raise LookupError('Missing attribute {} in file {}'.format(config_path, self.filename))
                 return default
-        return d
+        return config_pointer
 
-    def getList(self, name, default=[]):
-        return [i.strip() for i in self.get(name, ','.join(default)).split(',')]
+    def getList(self, config_path: str, default: list = []) -> list:
+        return [i.strip() for i in self.get(config_path, ','.join(default)).split(',')]
 
-    def getBoolean(self, name, default=False) -> bool:
-        value = self.get(name, default)
+    def getBoolean(self, config_path: str, default=False) -> bool:
+        value = self.get(config_path, default)
         if isinstance(value, bool):
             return value
-        return bool(re.match('^(true|on|1|yes)$', value, flags=re.IGNORECASE))
+        return bool(re.match('^(true|on|1|yes)$', str(value), flags=re.IGNORECASE))
 
-    def has(self, name):
+    def has(self, config_path: str) -> bool:
         try:
-            self.get(name)
+            self.get(config_path)
             return True
         except Exception:
             pass
         return False
 
 
+class PCWConfig():
```

Comment:
I think a method which actualy does this namespace lookup would be nice.

A wrapper for this:
```
type(default)(ConfigFile().get('{}/{}'.format(namespace_section, field),
        return type(e['default'])(cfg.get(
                                              ConfigFile().get('{}/{}'.format(section, field), default)))
```
So we have the same logic for all values where a namespace is involved.

Reply:
sorry didn't get you here . you pointing your comment to classname so I am not sure what method do you mean .

Reply:
```
[cleanup]
max-images-per-flavor = 1
[cleanup.namespace.XXX]
max-images-per-flavor = 3
```

Reply:
`PCWcfg.get(namespace, 'cleanup/max-images-per-flavor');`

Reply:
I still see small contradiction here . from one side we speaking about aligning everything to logic which would look for value in this places and in this order : 
1. `{feature}.namespace.{namespace}/property`  ( e.g. `cleanup.namespace.kernel-qa/max-images-per-flavor` )
2. `{feature}/property` ( e.g. `cleanup/max-images-per-flavor` ) 
3. `default/property` ( e.g. `default/max-images-per-flavor` ) 

On other side we have current implementation of `cfgGet` in Provider class where instead of step 3 we looking into special dictionary within this method . and I don't know how would be better to merge this to approaches ? and do we want actually to require to have **any** value defined for step 3 or in some cases we can be fine to not get any value at all ?

Reply:
Hm, I'm not sure about `3.`. I actually didn't had this on my mind. What if you have a name like `max-age` and then two features with this value... both getting the default from `default/max-age` ?
Can you point me to the case where we want this `default/*` ?

And yes, I would like to have **any** value with a specified default value, even if the default is `None`.